packages/core/src/Stores/contract-store.js

```diff
@@ -155,18 +161,70 @@ export default class ContractStore extends BaseStore {
     }
 
     updateBarriersArray(contract_info, is_dark_mode) {
+        const {
+            contract_type,
+            current_spot,
+            current_spot_high_barrier,
+            current_spot_low_barrier,
+            barrier,
+            high_barrier,
+            low_barrier,
+            status,
+        } = contract_info || {};
+        const main_barrier = this.barriers_array?.[0];
+        if (isAccumulatorContract(contract_info.contract_type)) {
+            // even though updateBarriersArray is called both in DTrader & C.Details pages,
+            // the below code will delay Accumulator barriers and their labels only in Contract Details.
+            if (
+                (this.cached_barriers_data.current_spot_high_barrier === current_spot_high_barrier ||
+                    this.cached_barriers_data.current_spot_low_barrier === current_spot_low_barrier ||
+                    this.cached_barriers_data.high_barrier === high_barrier ||
+                    this.cached_barriers_data.low_barrier === low_barrier) &&
+                this.cached_barriers_data.status === status
+            ) {
+                return;
+            }
+            setTimeout(
```

Comment:
Why do we need a timeout?

Reply:
the requirement is to delay the moment when barriers are updated as part of its animation on FE side, so that we can see a consecutive tick-barriers movement: "`tick A -> the barriers calculated based on the tick A -> tick B -> the barriers calculated based on the tick B`" instead of "`tick A displayed together with the barriers calculated based on the tick A -> tick B displayed together with the barriers calculated based on the tick B`".